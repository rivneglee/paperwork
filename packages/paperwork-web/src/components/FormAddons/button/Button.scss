.pwapp-form-button {
  width: 100%;
  box-sizing: border-box;
  margin: 4px 0;
}
