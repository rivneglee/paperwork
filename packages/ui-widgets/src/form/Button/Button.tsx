import React, { FunctionComponent, ReactElement, SyntheticEvent } from 'react';
import classNames from 'classnames';

interface Props {
  icon?: ReactElement;
  iconPlacement?: 'left' | 'right';
  onClick?: (e: SyntheticEvent) => void;
  type?: 'outlined' | 'text' | 'link';
  size?: 'xs' | 's' | 'm' | 'l' | 'xl';
  color?: 'primary' | 'secondary';
  shadow?: boolean;
  className?: string;
  disabled?: boolean;
}

const Button: FunctionComponent<Props> = ({
  children,
  onClick,
  shadow,
  type,
  color,
  size,
  className,
  disabled,
  icon,
  iconPlacement = 'left',
}) => (
  <button disabled={disabled} className={classNames(
    'pw-button',
    shadow && 'pw-button--shadow',
    type && `pw-button--${type}`,
    color && `pw-button--color-${color}`,
    size && `pw-button--size-${size}`,
    disabled && 'pw-button--disabled',
    icon && iconPlacement === 'right' && 'pw-button--reversed',
    className && className,
  )} onClick={onClick}>
    {icon && <span className={classNames(
      'pw-button__icon',
      `pw-button__icon--${iconPlacement}`,
    )}>{icon}</span>}
    <span>{children}</span>
  </button>
);

export default Button;
