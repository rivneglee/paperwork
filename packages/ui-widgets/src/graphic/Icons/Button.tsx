/* tslint:disable:max-line-length */
import * as React from 'react';

export default ({ className = '' }: { className?: string }) => (
  <svg className={`pw-icon ${className}`} viewBox="0 0 1070 1024">
    <path
      d="M1085.6448 524.151467H611.396267v-81.92h392.738133V118.920533H135.168v323.310934h104.6528v81.92H54.0672V38.775467h1031.5776z"
    ></path>
    <path
      d="M909.9264 1016.149333h-566.613333L92.16 751.4112l131.345067-114.2784 90.5216 70.997333V242.824533h216.6784V572.074667h379.153066v444.0064z m-76.458667-75.776V650.581333H460.253867V318.395733H382.1568v489.472l-14.5408 36.0448-146.568533-122.197333-44.578134 38.0928 178.244267 180.565333h478.685867z"
    ></path>
  </svg>
);
