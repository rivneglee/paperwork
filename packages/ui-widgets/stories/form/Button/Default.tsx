import React from 'react';

import { Button, Card, Icons } from '../../../src';

export default () => (
  <div style={{ background: '#ebeef1', padding: '2.4rem' }}>
    <Card header={<h3>Button types</h3>}>
      <Button>
        Default
      </Button>
      <Button shadow>
        Shadow
      </Button>
      <Button type="outlined">
        Outlined
      </Button>
      <Button type="text">
        Text
      </Button>
      <Button type="link">
        Link
      </Button>
    </Card>
    <Card header={<h3>Button states</h3>}>
      <Button disabled>
        Disabled
      </Button>
      <Button disabled type="outlined">
        Disabled
      </Button>
      <Button disabled type="text">
        Disabled
      </Button>
      <Button disabled type="link">
        Disabled
      </Button>
    </Card>
    <Card header={<h3>Button colours</h3>}>
      <Button color="primary">
        Primary
      </Button>
      <Button color="primary" type="outlined">
        Primary
      </Button>
      <Button color="primary" type="text">
        Primary
      </Button>
      <Button color="primary" type="link">
        Primary
      </Button>
      <Button color="secondary">
        Secondary
      </Button>
      <Button color="secondary" type="outlined">
        Secondary
      </Button>
      <Button color="secondary" type="text">
        Secondary
      </Button>
      <Button color="secondary" type="link">
        Secondary
      </Button>
      <Button color="danger">
        Danger
      </Button>
      <Button color="danger" type="outlined">
        Danger
      </Button>
      <Button color="danger" type="text">
        Danger
      </Button>
      <Button color="danger" type="link">
        Danger
      </Button>
    </Card>
    <Card header={<h3>Button sizes</h3>}>
      <Button color="primary" size="xs" type="outlined">
        XS
      </Button>
      <Button color="primary" size="s" type="outlined">
        S
      </Button>
      <Button color="primary" size="m" type="outlined">
        M
      </Button>
      <Button color="primary" size="l" type="outlined">
        L
      </Button>
      <Button color="primary" size="xl" type="outlined">
        XL
      </Button>
    </Card>
    <Card header={<h3>Button icons</h3>}>
      <Button color="primary" icon={<Icons.Chart />}>
        Left icon
      </Button>
      <Button color="primary" icon={<Icons.Form />} iconPlacement="right">
        Right icon
      </Button>
    </Card>
  </div>
);
