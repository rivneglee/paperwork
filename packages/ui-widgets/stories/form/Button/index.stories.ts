import { storiesOf } from '@storybook/react';

import Default from './Default';

storiesOf('Button', module)
  .add('default', Default);
