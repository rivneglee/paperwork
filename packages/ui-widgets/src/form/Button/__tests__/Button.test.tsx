import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';

import Button from '../Button';

describe('Button', () => {
  it('should render default button', () => {
    const wrapper: ShallowWrapper = shallow(
      <Button>Button</Button>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render outlined button', () => {
    const wrapper: ShallowWrapper = shallow(
      <Button type="outlined">Button</Button>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render text button', () => {
    const wrapper: ShallowWrapper = shallow(
      <Button type="text">Button</Button>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render link button', () => {
    const wrapper: ShallowWrapper = shallow(
      <Button type="text">Button</Button>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render primary color button', () => {
    const wrapper: ShallowWrapper = shallow(
      <Button color="primary">Button</Button>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render secondary color button', () => {
    const wrapper: ShallowWrapper = shallow(
      <Button color="secondary">Button</Button>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render disabled button', () => {
    const wrapper: ShallowWrapper = shallow(
      <Button disabled>Button</Button>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render button with left icon', () => {
    const wrapper: ShallowWrapper = shallow(
      <Button icon={<span>icon</span>}>Button</Button>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render button with right icon', () => {
    const wrapper: ShallowWrapper = shallow(
      <Button iconPlacement="right" icon={<span>icon</span>}>Button</Button>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
