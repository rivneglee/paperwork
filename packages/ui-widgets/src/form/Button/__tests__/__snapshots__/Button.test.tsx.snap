// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render button with left icon 1`] = `
<button
  className="pw-button"
>
  <span
    className="pw-button__icon pw-button__icon--left"
  >
    <span>
      icon
    </span>
  </span>
  <span>
    Button
  </span>
</button>
`;

exports[`Button should render button with right icon 1`] = `
<button
  className="pw-button pw-button--reversed"
>
  <span
    className="pw-button__icon pw-button__icon--right"
  >
    <span>
      icon
    </span>
  </span>
  <span>
    Button
  </span>
</button>
`;

exports[`Button should render default button 1`] = `
<button
  className="pw-button"
>
  <span>
    Button
  </span>
</button>
`;

exports[`Button should render disabled button 1`] = `
<button
  className="pw-button pw-button--disabled"
  disabled={true}
>
  <span>
    Button
  </span>
</button>
`;

exports[`Button should render link button 1`] = `
<button
  className="pw-button pw-button--text"
>
  <span>
    Button
  </span>
</button>
`;

exports[`Button should render outlined button 1`] = `
<button
  className="pw-button pw-button--outlined"
>
  <span>
    Button
  </span>
</button>
`;

exports[`Button should render primary color button 1`] = `
<button
  className="pw-button pw-button--color-primary"
>
  <span>
    Button
  </span>
</button>
`;

exports[`Button should render secondary color button 1`] = `
<button
  className="pw-button pw-button--color-secondary"
>
  <span>
    Button
  </span>
</button>
`;

exports[`Button should render text button 1`] = `
<button
  className="pw-button pw-button--text"
>
  <span>
    Button
  </span>
</button>
`;
